replica/table.cc

```diff
@@ -549,9 +549,21 @@ void table::enable_off_strategy_trigger() {
 
 storage_group_manager::~storage_group_manager() = default;
 
+future<> storage_group_manager::for_each_storage_group(const rwlock::holder&, std::function<future<>(size_t, storage_group&)> f) const {
```

Comment:
please add comment explaining that the caller should acquire a read/write lock since the function may yield.